cql3/expr/expression.hh

```diff
@@ -72,6 +72,23 @@ class unresolved_identifier;
 using expression = std::variant<bool, conjunction, binary_operator, column_value, column_value_tuple, token,
                                 unresolved_identifier>;
 
+// An expression variant element can't contain an expression by value, since the size of the variant
+// will be infinite. `nested_expression` contains an expression indirectly, but has value semantics and
+// is copyable.
+class nested_expression {
```

Comment:
nitpick: I think maybe the names `indirect_expression` or `expression_ptr` would be more self-explanatory than `nested_expression`. But I guess `nested_expression` isn't that bad, so I'll accept.

Reply:
expression_ptr implies pointer semantics (copying the pointer does not copy the pointee)